src/ucp/core/ucp_ep.c

```diff
@@ -2315,7 +2315,9 @@ void ucp_ep_do_keepalive(ucp_ep_h ep, ucp_lane_map_t *lane_map)
         status = uct_ep_check(ep->uct_eps[lane], 0, NULL);
         if (status == UCS_OK) {
             *lane_map &= ~UCS_BIT(lane);
-        } else if (status != UCS_ERR_NO_RESOURCE) {
+        } else if ((status != UCS_ERR_NO_RESOURCE) &&
+                   (status != UCS_ERR_ENDPOINT_TIMEOUT)) {
```

Comment:
maybe we need to return UCS_OK from UCT instead, so this ep would be immediately removed from the lane map?

Reply:
ok, returned success